docs/synapse/userguides/ug015_storm_ref_pivot.rst

```diff
@@ -11,85 +15,339 @@ Where specific query examples are given, they are commonly provided in pairs usi
 
 fromtags()
 ----------
-Todo
+Given a working set that contains one or more ``syn:tag`` nodes, returns the specified set of nodes to which those tags have been applied.
+
+``fromtags()`` can be thought of as pivoting **from** a set of **tags**, to a set of nodes that have those tags.
+
+Optional parameters:
+
+*  **<form>:** return only nodes of the specified form(s).
+* If no forms are specified, ``fromtags()`` returns all nodes for all forms to which the tags are applied.
+
+**Operator syntax:**
+
+.. parsed-literal::
+  
+  **fromtags(** [ *<form_1>* **,** *<form_2>* **,** *...<form_n>* ] **)**
+
+**Macro syntax:**
+
+N/A
+
+**Examples:**
+
+* Return the set of all nodes to which a given set of tags have been applied:
+  ::
+    fromtags()
+
+* Return the set of ``inet:fqdn`` and ``inet:email`` nodes to which a given set of tags have been applied:
+  ::
+    fromtags( inet:fqdn, inet:email )
+
+**Usage notes:**
+
+* ``fromtags()`` pivots from leaf tags only. For example, if the working set contains ``syn:tag=foo.bar.baz``, ``fromtags()`` will return nodes with ``#foo.bar.baz`` but **not** nodes with ``#foo.bar`` or ``#foo`` alone.
+* In some cases, pivoting with ``fromtags()`` is equivalent to lifting by tag; for example, ``ask #foo.mytag`` is equivalent to ``ask syn:tag=foo.mytag fromtags()``. However, ``fromtags()`` can also take more complex queries as input.
+* For example, say you are tagging nodes with analytical observations made by third parties: ``syn:tag=alias.acme.redtree`` ("things Acme Corporation states are associated with "Redtree" malware") or ``syn:tag=alias.foo.redtree`` ("things Foo Organization states are associated with "Redtree" malware"). To return all nodes **any** organization associates with "Redtree" you could do:
+  
+  ``ask syn:tag:base=redtree fromtags()``
+
+* ``totags()`` and ``fromtags()`` are often used together to:
+
+  * pivot from a set of nodes, to the tags applied to those nodes, to other nodes that have the same tags; or
+  * from a set of tags, to nodes those tags are applied to, to other tags applied to those same nodes.
 
 totags()
 --------
-Todo
+Returns the set of all tags (``syn:tag`` nodes) applied to the working set of nodes.
+
+``totags()`` can be thought of as pivoting from a set of nodes, **to** the set of **tags** applied to those nodes.
+
+Optional parameters:
+
+* **leaf:** specify whether ``totags()`` should return **only** leaf tags (``leaf = 1``) or **all** tags in the tag hierarchy (``leaf = 0``).
+
+* If no parameter is specified, ``totags()`` assumes ``leaf = 1``.
+
+**Operator syntax:**
+
+.. parsed-literal::
+  
+  **totags(** [ **leaf = 1** | **0** ] **)**
```

Comment:
`limit` does *not* work on totags

Reply:
Yerp. No changes to existing docs, although it's interesting to note that using the limit() *operator* after totags() also has no effect.

Reply:
https://github.com/vertexproject/synapse/pull/559

Reply:
I put up a PR which adds limit to the `totags` operator. I think we can merge this PR without changing the docs. I just think that `fromtags` and `totags` should behave the same way.